### DISCUSSION TITLE
How to use old XML schemes in maxGraph?

### DISCUSSION THREAD

**Discussion by K-A-A** [maxGraph/maxGraph#132] [Q&A] [answered]
I have a lot of schemes saved in XML by mxGraph.
Now I try to load them in maxGraph Graph Editor and they don't load.

I found that _packages\html\stories\stashed\grapheditor\www\js\Editor.js_ has method `setGraphXml`. It uses _mxGraphModel_ root element to start decoding (it's ok, old schemes have this root node, but new schema don't support them).
But... _packages\core\src\serialization\CodecRegistry.ts_ in maxGraph has another set of codecs (no _mxGraph_, _mxCell_ and others). I tried to remove _mx_ prefixes in my XML schemas and replace _mxGraphModel_ in `setGraphXml` (_Editor.js_) to _GraphModel_. And... It also doesn't work because _packages\core\src\serialization\Codec.ts_ in `decode` method use `ctor = window[node.nodeName];` to extract the appropriate codec. It is old code from mxGraph and it doesn't work.

Does anyone have any ideas on how to work with old XML schemas and how to convert them to new format?

**Comment by tbouffard** [comment, marked as answer]
Hi
Parts of this feature will be fixed when #125 will be merged
Please read https://github.com/maxGraph/maxGraph/pull/125#issuecomment-1287849617 and #102 for more information and to know some limitations.

This is a topic that must be addressed to ensure mxGraph users can reuse with `maxGraph` their past model created with `mxGraph`.

**Reply by ceisele-r** [reply]
Hi,
Is there anywhere an example how to load and display the graph data from `mxGraph` data correctly?

I tried the following but unfortunately it renders an empty chart:
```typescript
const doc = xmlUtils.parseXml(mxXmlString);
const codec = new Codec(doc);
const graph = new Graph(container);
codec.decode(doc.documentElement, graph.getDataModel());
```

**Reply by tbouffard** [reply]
Hi @ceisele-r 
The feature is not fully implemented in maxGraph and there is no working example.
To understand what is going on after the load, you can check the content of the model. You can have a look at #183
If possible open an issue describing your problem and share the xml model you use.
